packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -423,6 +424,19 @@ export class KubernetesClient {
     }
   }
 
+  // Check that we can connect to the cluster and return a Promise<boolean> of true or false depending on the result.
+  // We will check via trying to retrieve a list of API Versions from the server.
+  async checkConnection(): Promise<boolean> {
```

Comment:
maybe return a more complex object
because client might be interested in having the code

Reply:
I've opened up this issue here: https://github.com/containers/podman-desktop/issues/2204 to keep this PR simple and short, we can implement this in a different PR.